File: Ninja Slayground 2.0/10. Highest  Lowest Frequency Elements/Solution.py
#Learn the code, Copy and Paste is not allowed. You are required to type...

#Follow this account, for all future code updates on GitHub.

from typing import List
from collections import Counter

def getFrequencies(v: List[int]) -> List[int]:
    freq_map = Counter(v)
    max_freq = -1
    min_freq = float('inf')
    max_freq_element = None
    min_freq_element = None
    
    for num, freq in freq_map.items():
        if freq > max_freq or (freq == max_freq and num < max_freq_element):
            max_freq = freq
            max_freq_element = num
        if freq < min_freq or (freq == min_freq and num < min_freq_element):
            min_freq = freq
            min_freq_element = num
    
    return [max_freq_element, min_freq_element]

File: Ninja Slayground 2.0/10. Highest  Lowest Frequency Elements/Solution.java
// Learn the code, And copy and paste is not allowed. You are required to type...

// Follow this account, for all future code updates on GitHub.

import java.util.HashMap;
import java.util.Map;

public class Solution {
    public static int[] getFrequencies(int[] v) {
        Map<Integer, Integer> freqMap = new HashMap<>();
        
        for (int num : v) {
            freqMap.put(num, freqMap.getOrDefault(num, 0) + 1);
        }
        
        int maxFreq = -1;
        int minFreq = Integer.MAX_VALUE;
        int maxFreqElement = -1;
        int minFreqElement = -1;
        
        for (Map.Entry<Integer, Integer> entry : freqMap.entrySet()) {
            int num = entry.getKey();
            int freq = entry.getValue();
            
            if (freq > maxFreq || (freq == maxFreq && num < maxFreqElement)) {
                maxFreq = freq;
                maxFreqElement = num;
            }
            
            if (freq < minFreq || (freq == minFreq && num < minFreqElement)) {
                minFreq = freq;
                minFreqElement = num;
            }
        }
        
        return new int[] { maxFreqElement, minFreqElement };
    }
}
